ogc/query-with-cql-filters/src/main/java/com/esri/samples/query_with_cql_filters/QueryWithCqlFiltersController.java

```diff
@@ -0,0 +1,264 @@
+  @FXML private CheckBox timeExtentCheckBox;
+
+  private OgcFeatureCollectionTable ogcFeatureCollectionTable; // keep loadable in scope to avoid garbage collection
+
+  public void initialize() {
+
+    try {
+
+      // authentication with an API key or named user is required to access basemaps and other location services
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+      // create a map with the topographic basemap style
+      ArcGISMap map = new ArcGISMap(BasemapStyle.ARCGIS_TOPOGRAPHIC);
+
+      // set the map to the mapview
+      mapView.setMap(map);
+
+      // prepare UI for interaction
+      populateUiWithCqlQueriesAndDateValues();
+
+      // define strings for the service URL and collection id
+      // note that the service defines the collection id which can be accessed via OgcFeatureCollectionInfo.getCollectionId().
+      String serviceUrl = "https://demo.ldproxy.net/daraa";
+      String collectionId = "TransportationGroundCrv";
+
+      // create an OGC feature collection table from the service url and collection id
+      ogcFeatureCollectionTable = new OgcFeatureCollectionTable(serviceUrl, collectionId);
+
+      // set the feature request mode to manual
+      // in this mode, the table must be manually populated - panning and zooming won't request features automatically
+      ogcFeatureCollectionTable.setFeatureRequestMode(ServiceFeatureTable.FeatureRequestMode.MANUAL_CACHE);
+
+      // load the table
+      ogcFeatureCollectionTable.loadAsync();
+
+      // ensure the feature collection table has loaded successfully before creating a feature layer from it to display on the map
+      ogcFeatureCollectionTable.addDoneLoadingListener(() -> {
+        if (ogcFeatureCollectionTable.getLoadStatus() == LoadStatus.LOADED) {
+
+          // create a feature layer and set a renderer to it to visualize the OGC API features
+          var featureLayer = new FeatureLayer(ogcFeatureCollectionTable);
+          var simpleRenderer = new SimpleRenderer(new SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, ColorUtil.colorToArgb(Color.DARKMAGENTA), 3));
+          featureLayer.setRenderer(simpleRenderer);
+
+          // add the layer to the map
+          map.getOperationalLayers().add(featureLayer);
+
+          // zoom to the dataset extent
+          Envelope datasetExtent = ogcFeatureCollectionTable.getExtent();
+          if (datasetExtent != null && !datasetExtent.isEmpty()) {
+            mapView.setViewpointGeometryAsync(datasetExtent);
+          }
+
+          // create a query based on the current visible extent
+          QueryParameters visibleExtentQuery = new QueryParameters();
+          visibleExtentQuery.setGeometry(datasetExtent);
+          // set a limit of 3000 on the number of returned features per request, the default on some services could be as low as 10
+          visibleExtentQuery.setMaxFeatures(3000);
```

Comment:
this max features is set as the sample loads, so that only 3000 features display on the map: when hitting the query button, it will take the value in the text field and use that to reset the max features shown on the query parameters. this line of code isn't setting a limit that has to be carried throughout the sample workflow if that makes sense? it's designed to be rewritten/reset with any number. perhaps I can help explain this a bit more with a comment: does this help make this a bit clearer? 

```java
// set a limit of 3000 on the number of returned features when the sample loads

```

Reply:
@Rachael-E Yep that makes sense so it's working as intended as an initial limit but the user is free to enter a value higher than that. All good!